src/frontend/apps/desk/src/features/teams/components/Member/ModalRole.tsx

```diff
@@ -0,0 +1,119 @@
+import {
+  Button,
+  Modal,
+  ModalSize,
+  Radio,
+  RadioGroup,
+  VariantType,
+  useToastProvider,
+} from '@openfun/cunningham-react';
+import { t } from 'i18next';
+import { useState } from 'react';
+
+import { Box, Text, TextErrors } from '@/components';
+import { useAuthStore } from '@/features/auth';
+import { Access, Role, useUpdateTeamAccess } from '@/features/teams/api/';
+
+interface ModalRoleProps {
+  access: Access;
+  onClose: () => void;
+  teamId: string;
+}
+
+export const ModalRole = ({ onClose, teamId, access }: ModalRoleProps) => {
+  const [localRole, setLocalRole] = useState(access.role);
+  const { userData } = useAuthStore();
+  const { toast } = useToastProvider();
+  const {
+    mutate: updateTeamAccess,
+    error: errorUpdate,
+    isError: isErrorUpdate,
+  } = useUpdateTeamAccess({
+    onSuccess: () => {
+      toast(t('The role has been updated'), VariantType.SUCCESS, {
+        duration: 4000,
+      });
+      onClose();
+    },
+  });
+
+  const isOtherOwner =
+    access.role === Role.OWNER && userData?.id !== access.user.id;
+  const isNotAllowed = isOtherOwner;
+
+  return (
+    <Modal
+      isOpen
+      leftActions={
+        <Button color="secondary" fullWidth onClick={() => onClose()}>
+          {t('Cancel')}
+        </Button>
+      }
+      onClose={() => onClose()}
+      rightActions={
+        <Button
+          color="primary"
+          fullWidth
+          onClick={() => {
+            updateTeamAccess({
+              role: localRole,
+              teamId,
+              accessId: access.id,
+            });
+          }}
+          disabled={isNotAllowed}
+        >
+          {t('Validate')}
+        </Button>
```

Comment:
nit: if the role hasn't been updated, and the initial/default checked radio item has not changed, do we disable the button? to avoid submitting request if no change happened.

Reply:
I think it is ok for now, it is really in the edge case. You can open a issue if you find it is necessary.

Reply:
your call, let's ignore it